### PR TITLE
Storage layer refactor

### PR DESCRIPTION
**Pull request by wpride**
Fairly big refactor to get us out of static singleton storage hell. Hopefully this should clear up a good deal of the "Storage not registered" exceptions.

Primarily this overrides the InstallerFactory and *ResourceInstaller classes to access Formplayer's user-specific SQL storage factories instead of the static StorageManager's version. This code is also moved into factory beans.

At the CommCare level there are a few refactors to allow this, the biggest being the removal of the (as far as I can tell) useless CommCareInstance interface which allows us to access the storage layer through the platform when necessary (instead of the static helper). 

@benrudolph

### CODE REVIEW

**Comment by benrudolph** [comment]
(let me know when this is ready for review)

File: src/main/java/application/IncompleteSessionController.java [hunk 1]
```diff
@@ -20,8 +20,10 @@
 import session.FormSession;
 import util.Constants;
 
-import java.util.*;
-import java.util.concurrent.locks.Lock;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
```
Comment (benrudolph) [inline src/main/java/application/IncompleteSessionController.java]: nit: not sure why this file's imports changed

File: src/main/java/db/migration/V10__form_session_add_app_id.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.jdbc.core.JdbcTemplate;
+
+/**
+ * Add appId to form session
+ */
+public class V10__form_session_add_app_id implements SpringJdbcMigration {
+    @Override
+    public void migrate(JdbcTemplate jdbcTemplate) throws Exception {
+        jdbcTemplate.execute("ALTER TABLE formplayer_sessions " +
+                "ADD appid VARCHAR DEFAULT '1'");
```
Comment (benrudolph) [inline src/main/java/db/migration/V10__form_session_add_app_id.java]: should the default not by NULL so we don't get confused if there was an actual app associated with the form?

File: src/main/java/installers/FormplayerXFormInstaller.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        storageFactory = new FormplayerStorageFactory();
+        storageFactory.configure(databasePath, trimmedUsername);
+
+    }
+
+    @Override
+    public void writeExternal(DataOutputStream out) throws IOException {
+        super.writeExternal(out);
+        ExtUtil.writeString(out, ExtUtil.emptyIfNull(storageFactory.getDatabasePath()));
+        ExtUtil.writeString(out, ExtUtil.emptyIfNull(storageFactory.getTrimmedUsername()));
```
Comment (benrudolph) [inline src/main/java/installers/FormplayerXFormInstaller.java]: all these writeExternal and readExternal seem to be very similar, could make it slightly DRYer

File: src/main/java/objects/SerializableFormSession.java [hunk 4]
```diff
@@ -29,6 +27,7 @@
     private boolean oneQuestionPerScreen;
     private String asUser;
     private String currentIndex = "0";
+    private String appId;
```
Comment (benrudolph) [inline src/main/java/objects/SerializableFormSession.java]: 😂 😂 😂

File: src/main/java/application/IncompleteSessionController.java (hunk 1 above)
Comment (wpride) [inline src/main/java/application/IncompleteSessionController.java]: I just ran optimize imports across the source set

File: src/main/java/installers/FormplayerXFormInstaller.java (hunk 3 above)
Comment (wpride) [inline src/main/java/installers/FormplayerXFormInstaller.java]: Agreed - unfortunately the extant inheritance pattern and lack of multiple inheritance makes that slightly hard

**Pull request merged**